BuildResidentialHPXML/measure.rb

```diff
@@ -3232,304 +4671,553 @@ def arguments(model) # rubocop:disable Lint/UnusedMethodArgument
+      name: 'utility_bill_pv_feed_in_tariff_rates',
+      required: false,
+      display_name: 'Utility Bills: PV Feed-In Tariff Rates',
+      description: "Utility bill PV annual full/gross feed-in tariff rates. Only applies if the PV compensation type is '#{HPXML::PVCompensationTypeFeedInTariff}'. If multiple scenarios, use a comma-separated list."
+    )
+
+    args << makeArgument(
+      type: Argument::String,
+      name: 'utility_bill_pv_monthly_grid_connection_fee_units',
+      required: false,
+      display_name: 'Utility Bills: PV Monthly Grid Connection Fee Units',
+      description: 'Utility bill PV monthly grid connection fee units. If multiple scenarios, use a comma-separated list.'
+    )
+
+    args << makeArgument(
+      type: Argument::String,
+      name: 'utility_bill_pv_monthly_grid_connection_fees',
+      required: false,
+      display_name: 'Utility Bills: PV Monthly Grid Connection Fees',
+      description: 'Utility bill PV monthly grid connection fees. If multiple scenarios, use a comma-separated list.'
+    )
+
+    args << makeArgument(
+      type: Argument::String,
+      name: 'additional_properties',
+      required: false,
+      display_name: 'Additional Properties',
+      description: "Additional properties specified as key-value pairs (i.e., key=value). If multiple additional properties, use a |-separated list. For example, 'LowIncome=false|Remodeled|Description=2-story home in Denver'. These properties will be stored in the HPXML file under /HPXML/SoftwareInfo/extension/AdditionalProperties."
+    )
+
+    args << makeArgument(
+      name: 'combine_like_surfaces',
+      type: Argument::Boolean,
+      required: false,
+      display_name: 'Combine like surfaces?',
+      description: 'If true, combines like surfaces to simplify the HPXML file generated.'
+    )
+
+    args << makeArgument(
+      name: 'apply_defaults',
+      type: Argument::Boolean,
+      required: false,
+      display_name: 'Apply Default Values?',
+      description: 'If true, applies OS-HPXML default values to the HPXML output file. Setting to true will also force validation of the HPXML output file before applying OS-HPXML default values.'
+    )
+
+    args << makeArgument(
+      name: 'apply_validation',
+      type: Argument::Boolean,
+      required: false,
+      display_name: 'Apply Validation?',
+      description: 'If true, validates the HPXML output file. Set to false for faster performance. Note that validation is not needed if the HPXML file will be validated downstream (e.g., via the HPXMLtoOpenStudio measure).'
+    )
+
+    return args
+  end
 
+  # TODO
+  def defaultOptionalArgumentValues(args)
+    # these were previously required arguments with default values set (i.e., arg.setDefaultValue(xxx))
+    args[:floor_over_foundation_assembly_r] = 28.1 if args[:floor_over_foundation_assembly_r].nil?
+    args[:floor_over_garage_assembly_r] = 28.1 if args[:floor_over_garage_assembly_r].nil?
```

Comment:
I was really hoping we wouldn't need these (totally arbitrary) defaults. Why?

But if for some reason we *do* need them, this is definitely a step back from what we had before. Previously, at least a user opening the measure in the OS App or PAT would see the default value. Now they happen completely behind the scenes.

Reply:
Some, yes, are totally arbitrary. I can fix/refactor those.

Others, for example heat pump backup fuel, are not. We can report the default value in the argument description (there's already precedent for this) and set it in its respective `set_xxx` method.

Reply:
Why does heat pump backup fuel need a default?